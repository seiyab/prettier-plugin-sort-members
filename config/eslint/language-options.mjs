export function languageOptions(dir) {
	return {
		parserOptions: {
			projectService: true,
			tsConfigDirName: dir,
		},
	};
}
