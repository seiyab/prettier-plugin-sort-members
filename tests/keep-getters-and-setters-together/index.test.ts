import { describe, test, expect } from "bun:test";
import { readdir, readFile } from "node:fs/promises";
import { join } from "node:path";
import { format } from "prettier";
import { ESLint } from "eslint";

const plugins = ["./index.ts"];

const dir = join(import.meta.dir, "testdata");
const files = await readdir(dir);

describe("@typescript-eslint/keep-getters-and-setters-together", () => {
	describe.each([true, false])("%j", async (opt) => {
		test.each(files)("%s", async (name) => {
			const opts = {
				keepGettersAndSettersTogether: opt,
				sortMembersAlphabetically: true,
			};
			const path = join(dir, name);
			const code = await readFile(path, "utf-8");
			const result = await format(code, {
				...opts,
				filepath: path,
				plugins,
			});
			expect(result).toMatchSnapshot();
		});

		test.each(files)("parser agnostic: %s", async (name) => {
			const path = join(dir, name);
			const opts = {
				keepGettersAndSettersTogether: opt,
				sortMembersAlphabetically: true,
				filepath: path,
				plugins,
			};
			const code = await readFile(path, "utf-8");
			const result1 = await format(code, {
				...opts,
				parser: "typescript",
			});
			const result2 = await format(code, {
				...opts,
				parser: "babel-ts",
			});
			expect(result1).toBe(result2);
		});
	});

	test.each(files)("no conflict: %s", async (name) => {
		const eslint = new ESLint({
			overrideConfig: {
				parser: "@typescript-eslint/parser",
				plugins: ["@typescript-eslint"],
				extends: [],
				rules: {
					"@typescript-eslint/adjacent-overload-signatures": "error",
				},
			},
			useEslintrc: false,
		});
		const opts = {
			keepGettersAndSettersTogether: true,
			sortMembersAlphabetically: true,
		};
		const path = join(dir, name);
		const code = await readFile(path, "utf-8");
		const result = await format(code, {
			...opts,
			filepath: path,
			plugins,
		});

		const lintResults = await eslint.lintText(result);

		expect(lintResults).toHaveLength(1);
		expect(lintResults[0].messages).toBeEmpty();
	});
});
