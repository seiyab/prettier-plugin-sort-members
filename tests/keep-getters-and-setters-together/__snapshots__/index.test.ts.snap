// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`@typescript-eslint/keep-getters-and-setters-together keepGettersAndSettersTogether: true sortMembersAlphabetically: true getter-setter.ts 1`] = `
"export class TestClass {
  get theme(): string {
    return "light";
  }
  get visible(): boolean {
    return true;
  }
  set visible(_val: boolean) {}
}

class A {
  a: number;
  b: string;
  get c(): boolean {
    return true;
  }
  set c(_val: boolean) {}
  get d(): string {
    return "d";
  }
  set d(_val: string) {}
  get e(): boolean {
    return false;
  }
}
"
`;

exports[`@typescript-eslint/keep-getters-and-setters-together keepGettersAndSettersTogether: true sortMembersAlphabetically: true getter-setter.js 1`] = `
"export class TestClass {
  get theme() {}
  get visible() {}
  set visible(value) {}
}

class A {
  a = 1;
  d = 2;
  get b() {}
  set b(value) {}
  set c(value) {}
}
"
`;

exports[`@typescript-eslint/keep-getters-and-setters-together keepGettersAndSettersTogether: true sortMembersAlphabetically: false getter-setter.ts 1`] = `
"export class TestClass {
  get visible(): boolean {
    return true;
  }
  set visible(_val: boolean) {}
  get theme(): string {
    return "light";
  }
}

class A {
  a: number;
  b: string;
  get c(): boolean {
    return true;
  }
  get d(): string {
    return "d";
  }
  set c(_val: boolean) {}
  set d(_val: string) {}
  get e(): boolean {
    return false;
  }
}
"
`;

exports[`@typescript-eslint/keep-getters-and-setters-together keepGettersAndSettersTogether: true sortMembersAlphabetically: false getter-setter.js 1`] = `
"export class TestClass {
  get visible() {}
  get theme() {}
  set visible(value) {}
}

class A {
  a = 1;
  d = 2;
  get b() {}
  set c(value) {}
  set b(value) {}
}
"
`;

exports[`@typescript-eslint/keep-getters-and-setters-together keepGettersAndSettersTogether: false sortMembersAlphabetically: true getter-setter.ts 1`] = `
"export class TestClass {
  get theme(): string {
    return "light";
  }
  get visible(): boolean {
    return true;
  }
  set visible(_val: boolean) {}
}

class A {
  a: number;
  b: string;
  get c(): boolean {
    return true;
  }
  get d(): string {
    return "d";
  }
  get e(): boolean {
    return false;
  }
  set c(_val: boolean) {}
  set d(_val: string) {}
}
"
`;

exports[`@typescript-eslint/keep-getters-and-setters-together keepGettersAndSettersTogether: false sortMembersAlphabetically: true getter-setter.js 1`] = `
"export class TestClass {
  get theme() {}
  get visible() {}
  set visible(value) {}
}

class A {
  a = 1;
  d = 2;
  get b() {}
  set b(value) {}
  set c(value) {}
}
"
`;

exports[`@typescript-eslint/keep-getters-and-setters-together keepGettersAndSettersTogether: false sortMembersAlphabetically: false getter-setter.ts 1`] = `
"export class TestClass {
  get visible(): boolean {
    return true;
  }
  get theme(): string {
    return "light";
  }
  set visible(_val: boolean) {}
}

class A {
  a: number;
  b: string;
  get c(): boolean {
    return true;
  }
  get d(): string {
    return "d";
  }
  get e(): boolean {
    return false;
  }
  set c(_val: boolean) {}
  set d(_val: string) {}
}
"
`;

exports[`@typescript-eslint/keep-getters-and-setters-together keepGettersAndSettersTogether: false sortMembersAlphabetically: false getter-setter.js 1`] = `
"export class TestClass {
  get visible() {}
  get theme() {}
  set visible(value) {}
}

class A {
  a = 1;
  d = 2;
  get b() {}
  set c(value) {}
  set b(value) {}
}
"
`;
